### PR TITLE
fix: explore and navmap scene's thumbnails

### PR DESCRIPTION
**Pull request by pbosio**
try to use builder's scene thumbnail when there is no thumbnail set in scene's json.
fix scaling in explore and navmap toast so thumbnail do not stretch.
fix navmap toast not handling image download error. 

test link: https://explorer.decentraland.zone/branch/fix/explore-navmap-thumbnails/index.html?ENV=org&ENABLE_EXPLORE_HUD

### CODE REVIEW

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Helpers/UIHelpers/RawImageFillParent.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class RawImageFillParent : RawImage
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Helpers/UIHelpers/RawImageFillParent.cs:5]: Can we add a bit of documentation about what this helper class is supposed to do?

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapToastView.cs [hunk 2]
```diff
@@ -145,32 +149,30 @@ private void OnGotoClick()
...
-        Texture currentImage;
+        Texture2D currentImage;
         Coroutine downloadCoroutine;
 
         private IEnumerator Download(string url)
         {
+            scenePreviewLoadingSpinner.SetActive(true);
+            scenePreviewImage.texture = null;
+
             UnityWebRequest www = UnityWebRequestTexture.GetTexture(url);
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapToastView.cs:160]: Maybe not in the scope of this PR, but for the record, this should be using `AssetPromiseKeeper_Texture`.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseSceneCellView.cs [hunk 3]
```diff
@@ -81,13 +81,9 @@ void IMapDataView.SetMinimapSceneInfo(MinimapMetadata.MinimapSceneInfo info)
...
         {
-            string url = mapInfo.previewImageUrl;
-            if (string.IsNullOrEmpty(url))
-            {
-                url = MapUtils.GetMarketPlaceThumbnailUrl(mapInfo, 196, 143, 50);
-            }
-
-            FetchThumbnail(url);
+            FetchThumbnail(mapInfo.previewImageUrl,
+                onFetchFail: () => FetchThumbnail(MapUtils.GetMarketPlaceThumbnailUrl(mapInfo, 196, 143, 50),
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseSceneCellView.cs]: What these magic numbers are for? We can't use constants here?

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs [hunk 4]
```diff
@@ -1,30 +1,34 @@
...
 
-    public event Action<Sprite> OnThumbnailFetched;
+    public event Action<Texture2D> OnThumbnailFetched;
 
     UnityWebRequest thumbnailRequest = null;
     Texture2D thumbnailTexture;
-    Sprite thumbnail;
 
-    public void FetchThumbnail(string url)
+    public void FetchThumbnail(string url, Action onFetchFail)
```
Comment (BrianAmadori) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs:19]: A bit confused on why we are passing `onFetchFail` as an argument, but we have `OnThumbnailFetched` as a member event.

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapToastView.cs (hunk 2 above)
Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavmapToastView.cs:160]: Yeap, I was going to add that after this PR cause we are in a hurry to free these prefabs

File: unity-client/Assets/Scripts/MainScripts/DCL/Helpers/UIHelpers/RawImageFillParent.cs (hunk 1 above)
Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Helpers/UIHelpers/RawImageFillParent.cs:5]: added summary comment

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseSceneCellView.cs (hunk 3 above)
Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseSceneCellView.cs]: added constant for those numbers

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs (hunk 4 above)
Comment (pbosio) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ExploreHUD/Scripts/BaseCellView.cs:19]: renamed `OnThumbnailFetched` to `OnThumbnailSet` cause the event is triggered when the thumbnail is set and not necessary means that the thumbnail was fetched successfully

**Review by BrianAmadori — APPROVED** [review]

**Pull request merged**